src/pair.py

```diff
@@ -459,3 +459,64 @@ def __init__(self, nlist, default_r_cut=None, mode='none'):
             ),
         )
         self._add_typeparam(params)
+
+
+class ExpandedYukawa(pair.Pair):
+    r"""Expanded Yukawa pair potential.
+
+    Args:
+        nlist (hoomd.md.nlist.NeighborList): Neighbor list.
+        r_cut (float): Default cutoff radius :math:`[\mathrm{length}]`.
+        mode (str): Energy shifting/smoothing mode.
+
+    `ExpandedYukawa` is a shifted Yukawa potential. The discontinuity is shifted
+    based on a :math:`\Delta` that is the minimum interaction distance between
+    particles.
+
+    The `ExpandedYukawa` potential is:
+
+    .. math::
+
+        U(r) = \varepsilon \frac{\exp[-\kappa(r - \Delta)]}{r - \Delta}
+
+    Example::
+
+        nl = nlist.Cell()
+        exyuk = pair.ExpandedYukawa(default_r_cut=4.0, nlist=nl)
+        # particle types A interacting
+        exyuk.params[('A', 'A')] = dict(epsilon=1.0, kappa=1.0, delta=2.0)
+
+    .. py:attribute:: params
+
+        The `ExpandedYukawa` potential parameters. The dictionary has the following
+        keys:
+
+        * ``epsilon`` (`float`, **required**) - Energy parameter :math:`\epsilon`
+          :math:`[\mathrm{energy}]`
+        * ``kappa`` (`float`, **required**) - Scaling parameter :math:`\kappa`
+          :math:`[\mathrm{length}]^{-1}`
+        * ``delta`` (`float`, **required**) - Minimum interaction distance
+          :math:`\Delta` :math:`[\mathrm{length}]`
+
+        Type: :class:`~hoomd.data.typeparam.TypeParameter` [`tuple`
+        [``particle_type``, ``particle_type``], `dict`]
+
+    .. py:attribute:: mode
+
+        Energy shifting/smoothing mode: ``"none"``.
```

Comment:
This should allow all cutoff modes after you add the energy shifting to the evaluator, please copy from another class.